js/chat_app/src/userCreationForm/UserCreationForm.jsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import "./user_create_form.css";
+import Avatar from "@mui/material/Avatar";
+import { green } from "@mui/material/colors";
+
+export const UserCreationForm = ({ styleForm }) => {
+  const [usernameInput, setUsernameInput] = useState("");
+  const [passwordInput, setPasswordInput] = useState("");
+  const [token, setToken] = useState("");
+  const [users, setUsers] = useState([]);
+
+  useEffect(() => {
+    function getAdminId() {
+      fetch("/api/login", {
+        method: "POST",
+        headers: {
+          "Content-type": "application/json; charset=UTF-8",
+        },
+        body: JSON.stringify({
+          username: "admin",
+          password: "admin",
+        }),
+      })
+        .then((res) => {
+          return res.json();
+        })
+        .then((data) => {
+          setToken(data);
+          getUsers(data.authToken);
+        });
+    }
+    getAdminId();
+  }, []);
+
+  function createUser(adminToken, adminId, username, password) {
+    fetch("/api/users", {
+      method: "POST",
+      headers: {
+        "Content-type": "application/json",
+        "Auth-Token": adminToken,
+      },
+      body: JSON.stringify({
+        adminId: adminId,
+        username: username,
+        password: password,
+      }),
+    });
+    getUsers(adminToken);
```

Comment:
This is a serious problem - things have to be done in sequence, not in parallel.
```suggestion
    fetch(...).then(() => {
        getUsers(adminToken);
    })

```